File: Sort.java
/************************************************************************************
* Author     : Viam
* File       : Sort.java
* Size       : 4.00 KB
* Language   : Java
* Role       : Generate an array of random numbers and sort them on a ascending order.
************************************************************************************/

import org.apache.commons.lang.ArrayUtils; // Most important import

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Random;

public
class Sorter {
    private
    ArrayList<Integer> numbers;

    public
    static void main(String[] args)
    {
        int[] process = fillArray(20000, 0, 210000);
        sorter(process, 20000);
    }

    public
    static int[] sorter(int array[], int size)
    {

        int[] useArray = array;
        Integer[] newArray = ArrayUtils.toObject(useArray);
        List<Integer> arrayList = new ArrayList(Arrays.asList(newArray));
        List<Integer> storeList = new ArrayList(Arrays.asList(newArray));
        storeList.clear();

        int counter = 1;
        int minval = 0;
        int diffsize = size - 1;
        int actualVal = 0;
        int storeArray[] = new int[size];
        int removeIndex = 0;

        System.out.println((arrayList));
        for (int i = 0; i < size + diffsize; i++) {
            for (int n = 0; n < size - 1; n++) {
                if (arrayList.get(minval) < arrayList.get(counter)) {
                    actualVal = arrayList.get(minval);
                    // System.out.println((arrayList.get(minval)) + " Less than " + arrayList.get(counter));
                    counter = counter + 1;
                    removeIndex = minval;
                }
                else {
                    actualVal = arrayList.get(counter);
                    //  System.out.println((arrayList.get(counter)) + " Less than " + arrayList.get(minval));
                    minval = counter;
                    removeIndex = counter;
                    counter = counter + 1;
                }
            }

            size = size - 1;
            counter = 1;
            minval = 0;

            if (arrayList.size() == 1) {
                storeList.add(arrayList.get(0));
                int d = 0;
                int[] ints = new int[storeList.size()];
                for (Integer u : storeList) {
                    ints[d++] = u;
                }
                System.out.println(storeList);
                return ints;
            }
            storeList.add(actualVal);
            arrayList.remove(removeIndex);
        }
        int d = 0;
        int[] ints = new int[storeList.size()];
        for (Integer u : storeList) {
            ints[d++] = u;
        }

        return ints;
    }

    public
    static int randomNum(int lower, int upper)
    {
        Random rand = new Random();
        int randomNum = lower + rand.nextInt((upper - lower) + 1);
        return randomNum;
    }

    public
    static int[] fillArray(int size, int lowerBound, int upperBound)
    {
        int holdArray[] = new int[size];
        int count_num = 0;
        int rand = 0;

        for (int count = 0; count < holdArray.length; count++) {
            holdArray[count] = 0;
        }

        for (int count = 0; count < holdArray.length; count++) {

            rand = randomNum(lowerBound, upperBound);
            if (ArrayUtils.contains(holdArray, rand)) {
                count_num = count_num + 1;
                System.out.println("Already contains " + rand + " generating new value");
                while (ArrayUtils.contains(holdArray, rand)) {
                    rand = randomNum(lowerBound, upperBound);
                    count_num = count_num + 1;
                    System.out.println("Already contains " + rand + " generating new value");
                }
            }
            holdArray[count] = rand;
        }

        return holdArray;
    }
}

File: Sort.py
#/***************************************************************************************
#* Author     : Viam / TP037249
#* File       : Sort.py
#* Size       : 4.00 KB
#* Language   : Python 2.7.x
#* Role       : Generates an array of random numbers, checks for duplicate elements,
#*				if found, replaces them	and sorts them on a ascending order.
#* Acknowledgements : I am studying from the book "Learning Python" by Mark Lutz, 
#*					  A book worth every cent that I paid
#*					  http://www.oreilly.com/pub/au/446
#**************************************************************************************/

# Rudimentary imports, nothing fancy
import random

global randomnum # Scope needs to be available for all

#Not used now, this method uses a built in function, I've already created the manual one
'''
def randomNumber(lowerBound,upperBound):
	randomnum = randint(lowerBound,upperBound)
	return randomnum
'''

#Sample is not used because it is a built in python function to produce non-repeating numbers, instead I'll do it manually
def generateRandom(lowerBound,upperBound):
	#array = random.sample(range(lowerBound,upperBound),size) # sample method from "random" package allows for non-repeating random numbers. Make sure sample size is not larger than population
	rand = random.randrange(lowerBound,upperBound+1)
	return rand


def fillArray(size,lowerBound,upperBound):
	count =0
	array =[]
	for i in range(0,size):
		random = generateRandom(lowerBound,upperBound)
		if random in array:
			count = count +1
			print "Array element at index" , i , " Already contains", random , "generating new value"
			while random in array:
				random = generateRandom(lowerBound,upperBound)
				if random in array:
					print "Array element at index" , i , " Already contains", random , "generating new value"
					count = count+1
		array.append(random)
	print "Numbers were repeated a total of  ", count , " time(s), and had to be re-generated"
	return array
				


def sortArray(size,lowerBound,upperBound):
	useArray = fillArray(size,lowerBound,upperBound)
	print useArray
	arraysize = len(useArray)
	counter = 1
	minval = 0
	difsize = arraysize -1
	actualVal =0
	newArray =[]

#Nested for loop to go through each element, then let each element go through the whole array comparing
	for i in range(0,arraysize+difsize): # Interesting calculation here for the nested for loop
		for n in range(0,arraysize-1):
			if useArray[minval] < useArray[counter]:
				counter = counter +1
				actualVal = useArray[minval]
			else:
				minval = counter
				actualVal = useArray[counter]
				counter = counter +1

		arraysize = arraysize -1
		counter =1
		minval =0
		if i + arraysize ==i: # To push the final element left in the array, as there is no need to compare when there is only 1 element left
			newArray.extend(useArray) # Notice the extend, no need to use append as this is the last element
			return newArray
		newArray.append(actualVal) 
		useArray.remove(actualVal)
	return newArray

print sortArray(20,0,19)
